component/loki/source/kubernetes/kubernetes.go

```diff
@@ -40,6 +42,14 @@ type Arguments struct {
 
 	// Client settings to connect to Kubernetes.
 	Client commonk8s.ClientArguments `river:"client,block,optional"`
+
+	Clustering Clustering `river:"clustering,block,optional"`
+}
+
+// Clustering holds values that configure clustering-specific behavior.
+type Clustering struct {
```

Comment:
@tpaschalis Did you have a specific idea for how to deduplicate this work? 

I'm happy to do it in this PR if you want to see it but I'm also worried about blocking this with more rounds of review.

Reply:
I think this type should probably reside in the cluster service package.

I'm not against doing it on this PR, and it sounds like an easy-enough change so that it won't be a blocker.